### PR TITLE
Implement JSON editor component

### PR DESCRIPTION
**Pull request by marjan-georgiev**
Adds `ngx-json-editor` component that allows editing a JSON object based on a json-schema.
Validates the json against the schema using ajv.

### CODE REVIEW

**Comment by metalex9** [comment]
I was getting an error when trying to build the demo:
```
ERROR in node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
```

Googling lead me to [this](https://github.com/angular/angular/issues/24165) and [this](https://github.com/angular/angular/issues/21670) which suggested I add `"@angular/*": ["node_modules/@angular/*"]` to `paths` in `tsconfig.json`. This fixed the issue.

Will merge when CI passes.

File: src/app/app.component.ts [hunk 1]
```diff
@@ -3,17 +3,17 @@ import { HashLocationStrategy, Location, LocationStrategy } from '@angular/commo
...
+import { Hotkey, HotkeysService } from '../../projects/swimlane/ngx-ui/src/lib/components/hotkeys';
+import { LoadingService } from '../../projects/swimlane/ngx-ui/src/lib/components/loading';
+import { NotificationService } from '../../projects/swimlane/ngx-ui/src/lib/components/notification';
+import { IconRegisteryService } from '../../projects/swimlane/ngx-ui/src/lib/services/icon-registery.service';
+import { InjectionService } from '../../projects/swimlane/ngx-ui/src/lib/services/injection.service';
 import { getComputedStyle, rgb2hex } from './app.utils';
-import { version, dependencies } from '../../dist/swimlane/ngx-ui/package.json';
+import { version, dependencies } from '../../projects/swimlane/ngx-ui/package.json';
 
 console.log(JSON.stringify(dependencies));
```
Comment (Hypercubed) [inline src/app/app.component.ts]: Console log....

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/schema-validator.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export class SchemaValidator {
+  /**
+   * Validates schemas of a specified type
+   */
+  static validate(schema: any, model: any): any[] {
+    const ajv = new Ajv({
+      allErrors: true
+    });
+    const valid = ajv.validate(schema, model);
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/schema-validator.ts]: Getting an error here if the schema is undefined.

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor.component.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+})
+export class JsonEditorComponent {
+  @Input()
+  model: any;
+
+  @Input()
+  schema: any;
+
+  @Input()
+  label: string;
```
Comment (Hypercubed) [inline projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor.component.ts]: I'm seeing an error if label is undefined.

**Pull request merged**